base/show.jl

```diff
@@ -90,7 +91,7 @@ function show_default(io::IO, x::ANY)
     nf = nfields(t)
     if nf != 0 || t.size==0
         if !show_circular(io, x)
-            recur_io = IOContext(io, :SHOWN_SET => x)
+            recur_io = IOContext(io, SHOWN_SET=x, multiline=false)
```

Comment:
I'm not sure that we really want to make 2x the number of IOContexts. Also, unlike the Pair constructor form, this form won't be stack allocatable and generates a bit more garbage to create.